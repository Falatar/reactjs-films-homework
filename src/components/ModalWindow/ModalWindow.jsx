import React, { Component } from 'react';
import { connect } from 'react-redux';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import style from './ModalWindow.scss';
import endSession from './actions';

class ModalWindow extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isShowed: false,
    };
  }

  show = () => {
    this.setState((state) => ({
      isShowed: !state.isShowed,
    }));
  }

  hide = () => {
    const { closeModal } = this.props;
    closeModal();
  }

  render() {
    const { status, root } = this.props;
    const { isShowed } = this.state;
    if (status && root.results !== undefined) this.show();
    return (
      <div className={classNames(style.Bord, isShowed ? style.Up : style.Down)}>
        <iframe src={isShowed ? `https://www.youtube.com/watch?v=${root.results[0].key}` : ''} title="trailer" />
        <div className={style.BotLain}>
          <button type="button" className={style.CloseButton} onClick={this.hide}>close</button>
        </div>
      </div>
    );
  }
}

ModalWindow.defaultProps = {
  status: false,
  root: {},
  closeModal: () => {},
};

ModalWindow.propTypes = {
  status: PropTypes.bool,
  root: PropTypes.objectOf(PropTypes.any),
  closeModal: PropTypes.func,
};

const mapStateToProps = (store) => ({
  status: store.modalReducer.status,
  root: store.modalReducer.root,
});

const mapDispatchToProps = (dispatch) => ({
  closeModal: () => dispatch(endSession()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ModalWindow);
