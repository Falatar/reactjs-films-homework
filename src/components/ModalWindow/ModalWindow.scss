.up {
    z-index: 200;
}

.down {
    z-index: -1;
}

.bord {
    position: fixed;
    height: 82%;
    width: 80%;
    left: 10%;
    top: 2.5%;
    bottom: 2.5%;
    border: 5px solid;
    border-color: deepskyblue;
    border-radius: 5px;
    padding: 5px;
    background-color: lightgrey;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.video {
    width: 90%;
    height: 94%;
}

.bot__lain {
    width: 100%;
    height: 6%;
    display: flex;
    flex-direction: row-reverse;
    margin: 1%;

    .close__button {
        height: 100%;
        width: 8%;
        background-color: deepskyblue;
        border-radius: 10px;
        color: white;
        font-size: 1.3em;
        outline: none;
        border: none;
        text-shadow: 0px 2px 0px #0098cc;
    }

    .close__button:active {
        background-image: -webkit-linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
        background-image: -moz-linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
        background-image: -o-linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
        background-image: -ms-linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
        background-image: linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
    }
}

.loading {
    color: deepskyblue;
    font-size: 2em;
}