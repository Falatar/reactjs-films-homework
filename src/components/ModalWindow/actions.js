const closeModal = () => async (dispatch) => dispatch({
  type: 'HIDE_MODAL_WINDOW',
  payload: false,
});

export default closeModal;
