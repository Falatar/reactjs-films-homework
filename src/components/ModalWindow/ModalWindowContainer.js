import { connect } from 'react-redux';
import { closeModal } from '../../modules/singleFilmModule/singleFilmModuleActions';
import { isModalActive, getVideoLink } from '../../modules/singleFilmModule/singleFilmModuleSelector';
import ModalWindow from './ModalWindow';

export const mapStateToProps = (store) => ({
  status: isModalActive(store),
  root: getVideoLink(store),
});

export const mapDispatchToProps = (dispatch) => ({
  endModalSession: () => dispatch(closeModal()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ModalWindow);
