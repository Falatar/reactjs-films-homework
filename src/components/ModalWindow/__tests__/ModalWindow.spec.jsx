import React from 'react';
import renderer from 'react-test-renderer';
import ModalWindow from '../ModalWindow';

describe('ModalWindow', () => {
  const root = {
    results: [
      {
        key: 'number',
      },
    ],
  };

  it('renders correctly in active mode', () => {
    const tree = renderer
      .create(<ModalWindow status root={root} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly in passive mode', () => {
    const tree = renderer
      .create(<ModalWindow status={false} root={root} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly in active mode without input', () => {
    const tree = renderer
      .create(<ModalWindow status root={{}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly in passive mode without input', () => {
    const tree = renderer
      .create(<ModalWindow status={false} root={{}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
