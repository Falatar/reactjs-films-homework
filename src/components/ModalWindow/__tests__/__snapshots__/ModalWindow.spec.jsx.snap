// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWindow renders correctly in active mode 1`] = `
<div
  className="bord up"
>
  <iframe
    allowFullScreen={true}
    className="video"
    frameBorder="0"
    src="https://www.youtube.com/embed/number"
    title="trailer"
  />
  <div
    className="bot__lain"
  >
    <button
      className="close__button"
      onClick={[Function]}
      type="button"
    >
      close
    </button>
  </div>
</div>
`;

exports[`ModalWindow renders correctly in active mode without input 1`] = `
<div
  className="bord up"
>
  <span
    className="loading"
  >
    Loading...
  </span>
</div>
`;

exports[`ModalWindow renders correctly in passive mode 1`] = `
<div
  className="bord down"
>
  <iframe
    allowFullScreen={true}
    className="video"
    frameBorder="0"
    src=""
    title="trailer"
  />
  <div
    className="bot__lain"
  >
    <button
      className="close__button"
      onClick={[Function]}
      type="button"
    >
      close
    </button>
  </div>
</div>
`;

exports[`ModalWindow renders correctly in passive mode without input 1`] = `
<div
  className="bord down"
>
  <span
    className="loading"
  >
    Loading...
  </span>
</div>
`;
